ddtrace/contrib/internal/vertexai/_utils.py

```diff
@@ -0,0 +1,256 @@
+import sys
+
+from vertexai.generative_models import GenerativeModel
+from vertexai.generative_models import Part
+
+from ddtrace.internal.utils import get_argument_value
+from ddtrace.llmobs._integrations.utils import tag_request_content_part
+from ddtrace.llmobs._integrations.utils import tag_response_part
+
+
+class BaseTracedVertexAIStreamResponse:
+    def __init__(self, generator, integration, span, is_chat):
+        self._generator = generator
+        self._dd_integration = integration
+        self._dd_span = span
+        self._chunks = []
+        self.is_chat = is_chat
+
+
+class TracedVertexAIStreamResponse(BaseTracedVertexAIStreamResponse):
+    def __enter__(self):
+        self._generator.__enter__()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._generator.__exit__(exc_type, exc_val, exc_tb)
+
+    def __iter__(self):
+        try:
+            for chunk in self._generator.__iter__():
+                # only keep track of first chunk for chat messages
```

Comment:
worth just elaborating why it's special cased for the chat messages (that the chunk gets mutated as it goes)